git/git.go

```diff
@@ -84,19 +85,116 @@ type CommitSummary struct {
 	Subject        string
 }
 
+// Prepend Git config instructions to disable Git LFS filter
+func gitConfigNoLFS(args ...string) []string {
+	// Before git 2.8, setting filters to blank causes lots of warnings, so use cat instead (slightly slower)
+	// Also pre 2.2 it failed completely. We used to use it anyway in git 2.2-2.7 and
+	// suppress the messages in stderr, but doing that with standard StderrPipe suppresses
+	// the git clone output (git thinks it's not a terminal) and makes it look like it's
+	// not working. You can get around that with https://github.com/kr/pty but that
+	// causes difficult issues with passing through Stdin for login prompts
+	// This way is simpler & more practical.
+	filterOverride := ""
+	if !Config.IsGitVersionAtLeast("2.8.0") {
+		filterOverride = "cat"
+	}
+
+	return append([]string{
+		"-c", fmt.Sprintf("filter.lfs.smudge=%v", filterOverride),
+		"-c", fmt.Sprintf("filter.lfs.clean=%v", filterOverride),
+		"-c", "filter.lfs.process=",
+		"-c", "filter.lfs.required=false",
+	}, args...)
+}
+
+// Invoke Git with disabled LFS filters
+func gitNoLFS(args ...string) *subprocess.Cmd {
+	return subprocess.ExecCommand("git", gitConfigNoLFS(args...)...)
+}
+
+func gitNoLFSSimple(args ...string) (string, error) {
+	return subprocess.SimpleExec("git", gitConfigNoLFS(args...)...)
+}
+
+func gitNoLFSBuffered(args ...string) (*subprocess.BufferedCmd, error) {
+	return subprocess.BufferedExec("git", gitConfigNoLFS(args...)...)
+}
+
+// Invoke Git with enabled LFS filters
+func git(args ...string) *subprocess.Cmd {
+	return subprocess.ExecCommand("git", args...)
+}
+
+func gitSimple(args ...string) (string, error) {
+	return subprocess.SimpleExec("git", args...)
+}
+
+func gitBuffered(args ...string) (*subprocess.BufferedCmd, error) {
+	return subprocess.BufferedExec("git", args...)
+}
+
+func CatFile() (*subprocess.BufferedCmd, error) {
+	return gitNoLFSBuffered("cat-file", "--batch-check")
+}
+
+func DiffIndex(ref string, cached bool) (*bufio.Scanner, error) {
+	args := []string{"diff-index", "-M"}
+	if cached {
+		args = append(args, "--cached")
+	}
+	args = append(args, ref)
+
+	cmd, err := gitBuffered(args...)
+	if err != nil {
+		return nil, err
+	}
+	if err = cmd.Stdin.Close(); err != nil {
+		return nil, err
+	}
+
+	return bufio.NewScanner(cmd.Stdout), nil
+}
+
+func HashObject(by []byte) (string, error) {
```

Comment:
What do you think about changing this to accept a reader? This will allow it to take larger objects without being forced to read it into memory first. Then you can get rid of the `bytes.NewReader()` in here.

I realize that the usage in `commands/command_pointer.go` will be just fine buffering in memory. But, I think it makes sense to change this function since it's now an exported `git` package function.